### ISSUE TITLE
change(release): Enable all user features in the `experimental` build

### ISSUE THREAD

**Issue by teor2345** [ZcashFoundation/zebra#7892]
### Motivation

We want to be able to point users to an experimental build for all new user features, not just one or two features we've been focused on recently. This makes using those features easier for users, because they don't have to build Zebra themselves.

It also makes our documentation simpler, because we can just say "use the `latest-experimental` image", rather than explaining which features are in it.

### Suggested Solution:

First PR:
- [ ] Add an `experimental` feature to `zebrad` that just activates `default-release-binaries`
- [ ] Admin: Set the [`RUST_EXPERIMENTAL_FEATURES` GitHub Actions variable](https://github.com/ZcashFoundation/zebra/settings/variables/actions) to `experimental`

Second PR:
- [ ] Add all user features to the `experimental` feature: `elasticsearch`, `journald`, `prometheus`, `filter-reload`, `shielded-scan`
- [ ] Add a config test for the `experimental` feature (as well as `getblockchain-rpcs` and `shielded-scan`), and add a config with all the experimental features enabled

#### Optional

If the `Test all` job is slow:

<details>

<summary>Not needed any more - job is fast</summary>

- [ ] Split it into an experimental job and a default features job
https://github.com/ZcashFoundation/zebra/blob/afbe80706092de2287417bd5cab248a9f8167065/.github/workflows/ci-unit-tests-docker.yml#L122-L125
- [ ] Consider building an experimental image so that we get build caching for that image
- [ ] Admin: Add branch protection rules for new jobs

</details>

### Complex Code or Requirements

We don't want to make CI fail when an admin changes the settings, so it's better to split this ticket into 2 PRs.

### Testing

Make sure the tests are actually running in CI.

**Comment by teor2345** [comment]
This is blocked by #7891 because we want the new features to appear in the new tags, after the breaking change.

**Comment by teor2345** [comment]
I've updated this ticket, it looks like we'll need an admin for parts of it, so let's check when @gustavovalverde is around.

**Comment by mpguerra** [comment]
@gustavovalverde I'm unscheduling this one because there are more pressing things this sprint and I'm a bit lost with this one... Is there anything left to do here?

**Comment by arya2** [comment]
> Is there anything left to do here?

The experimental build was added to CI in #8175 and is testing every PR update: https://github.com/ZcashFoundation/zebra/actions/runs/7997001572/job/21840879753#step:5:105

Some tests fail with the `elasticsearch` feature (see https://github.com/ZcashFoundation/zebra/pull/8175#discussion_r1470413330), so it was omitted for now.

We may want an issue for fixing those test failures and adding it to the experimental build, or for adding the `internal-miner` feature to the experimental build when it's restored.

**Comment by mpguerra** [comment]
Going to close this as the original scope was done. We should open follow up issues if we want to add any new optional features, such as `elasticsearch` or `internal-miner` to the experimental build
